articles/tutorials/advanced/2d_shaders/05_transition_effect/index.md

```diff
@@ -0,0 +1,528 @@
+
+Now you can control the edge width slider to see the smooth edge between transitioned and not. 
+![Figure 5.9: A smooth edge](./gifs/smoothstep.gif)
+
+After we find an `EdgeWidth` value that looks good, we can set it in C# after the `SceneTransitionMaterial` is loaded.
+```csharp
+SceneTransitionMaterial.SetParameter("EdgeWidth", .05f);
+```
+
+>[!warning] 
+> Shader parameters do not use initializer expressions.
+>
+> If you set a default expression for a shader parameter, like setting `EdgeWidth=.05`, MonoGame's shader compiler ignores the `=.05` part. You will always need to set this value from C#.  
+
+### More Interesting Wipes
+
+So far the shader has been using `uv.x` to create a horizontal screen wipe. It would be easy to use `uv.y` to create a vertical screen wipe. 
+
+```hlsl
+float transitioned = smoothstep(Progress, Progress + EdgeWidth, uv.y);
+```
+
+![Figure 5.10: A vertical screen wipe](./gifs/vertical.gif)
+
+But what if we wanted to create more complicated wipes that didn't simply go in one direction? So far, we have passed `uv.x` and `uv.y` along as the argument to compare against the `Progress` shader parameter, but we could use any value we wanted. 
+
+Pull out the expression into a separate variable, `value`, and experiment with some different mathematical functions. For example, here is a wipe that comes in from the left and right towards the center.
+```hlsl
+float value = 1 - abs(.5 - uv.x) * 2;  
+float transitioned = smoothstep(Progress, Progress + EdgeWidth, value);
+```
+![Figure 5.11: A more interesting wipe](./gifs/center-wipe.gif)
+
+That is cool, but if we wanted an even more interesting wipe, the math would start to become challenging. In the final effect, it would also be nice to change the _type_ of wipe dynamically from the game, and having to change entire shader functions would be cumbersome. We can build a more generalized approach instead of writing mathematical functions to encode the wipe's progress. 
+
+To build intuition, we can visualize _just_ the `value` that is compared against the `Progress` parameter. 
+
+```hlsl
+float value = 1 - abs(.5 - uv.x) * 2;  
+return float4(value, value, value, 1);
+```
+
+![Figure 5.12: Just the value for the center wipe](./images/center-wipe-value.png)
+
+That image is not very interesting in of itself, but it does convey meaning about the transition effect _will_ render later. The darker areas of the image are going to transition sooner than the brighter areas, and the brightest areas will be the _last_ areas to transition when the `Progress` parameter is set all the way to `1`. At the end of the day, the image is just a grayscale gradient.
+
+You could imagine _other_ grayscale gradient images. In fact, there is a fantastic pack of _free_ gradient images made by _Screaming Brain Studios_ on [opengameart.org](https://opengameart.org/content/300-gradient-textures). Here are few samples, 
+
+![Figure 5.13: A sample gradient texture](./images/assets/ripple.png)
```

Comment:
Put the four images in a table side by side to take up less space on screen
Maybe with a second row for each column explaining what each item is meant to look like or the effct it will generate.